packages/elementor-ui/components/utils.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * @param styles - { base: { shared: '', variant: { h1: '', h2: '' } }, dark: { shared: '', variant: { h1: '', h2: '' } } }
+ * @param config - { variants: { light: true, dark: false } }
+ * @param props - { variant: 'h1', size: 'xl' }
+ * @returns {string}
+ */
+export const getStyle = ( styles, { props, config } ) => {
+	if ( ! styles ) {
+		return '';
+	}
+
+	if ( 'string' === typeof styles ) {
+		return styles;
+	}
+
+	// Creating an array that holds only the active theme variants values.
+	const themeVariants = Object.keys( config.variants ).filter( ( key ) => config.variants[ key ] );
+
+	// Adding the 'base' key to be included as the first variant.
+	themeVariants.unshift( 'base' );
+
+	let variantStyle = '';
+
+	themeVariants.forEach( ( key ) => {
+		const themeVariant = styles[ key ];
+
+		// If key exist in the styles obj (dark, light etc.)
+		if ( themeVariant ) {
+			if ( 'string' === typeof themeVariant ) {
+				variantStyle += themeVariant;
+			} else {
+				variantStyle += themeVariant.shared || '';
+			}
+
```

Comment:
Try making it an inner function the can be reused below

Reply:
Done.